clean parametric objects by default

akaszynski opened the pull request:
Our parametric objects are not manifold, and it's leading to some plotting artifacts when using `smooth_shading=True`.  Normally this isn't too much to worry about, but when working on updates for `tetgen`, it's causing one of the examples to fail.

This PR introduces a simple flag (default `True`) to enable automatic point merging of parametric objects.  For example:

```py
import pyvista as pv

pv.set_plot_theme('document')
pseudosphere = pv.ParametricPseudosphere()
pseudosphere.plot(smooth_shading=True)
```
![sc0](https://user-images.githubusercontent.com/11981631/133953646-c1acbf92-32dc-441a-b2ac-ba3854ae828a.png)


When disabled (default behavior prior to this PR):

```py
pseudosphere = pv.ParametricPseudosphere(clean=False)
pseudosphere.plot(smooth_shading=True)
```
![sc1](https://user-images.githubusercontent.com/11981631/133953649-59b5eb50-05b7-44e5-92aa-87d6738a9394.png)

adeak commented:
Linked https://github.com/pyvista/pyvista/issues/1301. I'd been wondering if a fixed tolerance is safe. I have very dense or very large meshes in mind.

akaszynski commented:
> Linked #1301. I'd been wondering if a fixed tolerance is safe. I have very dense or very large meshes in mind.

Quite right, fixed tolerance isn't the way to go.  I've changed it to relative tolerance.

The "clean" step comes at a cost, so I've disabled it for parametric geometry that isn't expected to be manifold.
